File: src/test/java/com/alibaba/mtc/testmodel/Task.java
package com.alibaba.mtc.testmodel;

import com.alibaba.mtc.MtContextThreadLocal;
import com.alibaba.mtc.Utils;

import java.util.Map;
import java.util.concurrent.ConcurrentMap;

/**
 * @author ding.lid
 */
public class Task implements Runnable {
    public final String tag;
    private ConcurrentMap<String, MtContextThreadLocal<String>> mtContexts;

    public Task(String tag, ConcurrentMap<String, MtContextThreadLocal<String>> mtContexts) {
        this.tag = tag;
        this.mtContexts = mtContexts;
    }

    public volatile Map<String, Object> copied;

    @Override
    public void run() {
        try {
            copied = Utils.modifyMtContexts(tag, mtContexts);
        } catch (Throwable e) {
            e.printStackTrace();
        }
    }
}

File: library/src/test/java/com/alibaba/ttl/testmodel/Task.kt
package com.alibaba.ttl.testmodel

import com.alibaba.createChildTtlInstancesAndModifyParentTtlInstances
import com.alibaba.ttl.TransmittableThreadLocal
import java.util.concurrent.ConcurrentHashMap
import java.util.concurrent.ConcurrentMap

/**
 * @author Jerry Lee (oldratlee at gmail dot com)
 */
class Task(private val tag: String, private val ttlInstances: ConcurrentMap<String, TransmittableThreadLocal<String>> = ConcurrentHashMap()) : Runnable {

    lateinit var copied: Map<String, String>

    override fun run() {
        copied = createChildTtlInstancesAndModifyParentTtlInstances(tag, ttlInstances)
    }
}
